app/Post_Types/Archive_Item/Rest.php

```diff
@@ -0,0 +1,343 @@
+			'get_callback' => function( $arr ) {
+				return get_the_permalink($arr['id']);
+			}
+		) );
+
+		// only do this extra if the popup is used.
+		$has_popup = carbon_get_theme_option(Settings::FIELD_GENERAL_BROWSE_MODAL);
+		if (!$has_popup) {
+			return;
+		}
+		// error_log(print_r( $this->fields, true ) , 3, "/Applications/MAMP/logs/php_error.log");
+
+		// Taxonomies are already added
+
+		// add CPT field vars
+		$fields = $this->get_fields();
+		$cpt_fields = array_filter($fields, function($field) {
+			return $field[self::FIELD_INDEX_TYPE] === CPT_Field::NAME;
+		});
+		if ($cpt_fields) {
+			register_rest_field( Archive_Item::NAME, 'cpts', array(
+				'get_callback' => function( $arr ) use( &$cpt_fields) {
+					$ret = [];
+					// error_log(print_r( $cpt_fields, true ) , 3, "/Applications/MAMP/logs/php_error.log");
+					foreach($cpt_fields as $field) {
+						$field_id = carbon_get_post_meta( $field[self::FIELD_INDEX_ID],FieldPostMeta::FIELD_ID );
+						// $ret[$field_cpt_id] = carbon_get_post_meta( $arr['id'],FieldPostMeta::FIELD_ID );
+
+						// $type = carbon_get_post_meta( $cptid, Post_Meta::FIELD_TYPE );
+						$ret[$field_id] = carbon_get_post_meta( $arr['id'], $field_id);
+
+					}
+					return $ret;
+				}
+			) );
+		}
+
+		// add Selects field vars
+		$select_fields = array_filter($fields, function($field) {
+			return $field[self::FIELD_INDEX_TYPE] === Select_Field::NAME;
+		});
+		if ($select_fields && count($select_fields)) {
+			register_rest_field( Archive_Item::NAME, 'selects', array(
+				'get_callback' => function( $arr ) use( &$select_fields) {
+					$ret = [];
+					foreach($select_fields as $field) {
+						$field_id = carbon_get_post_meta( $field[self::FIELD_INDEX_ID],FieldPostMeta::FIELD_ID );
+						$ret[$field_id] = carbon_get_post_meta( $arr['id'], $field_id);
+
+					}
+					return $ret;
+				}
+			) );
+		}
+
+		$text_fields = array_filter($fields, function($field) {
+			return $field[self::FIELD_INDEX_TYPE] === Text_Field::NAME;
+		});
+		if ($text_fields && count($text_fields)) {
+			register_rest_field( Archive_Item::NAME, 'fields_text', array(
+				'get_callback' => function( $arr ) use( &$text_fields) {
+					$ret = [];
+					foreach($text_fields as $field) {
+						$field_id = carbon_get_post_meta( $field[self::FIELD_INDEX_ID],FieldPostMeta::FIELD_ID );
+						$ret[$field_id] = carbon_get_post_meta( $arr['id'], $field_id);
+					}
+					return $ret;
+				}
+			) );
+		}
+
+		register_rest_field( Archive_Item::NAME, 'test', array(
+			'get_callback' => function( $arr ) use( &$text_fields) {
+				return count($text_fields);
+			}
+		) );
+
+	}
+
+	public function get_image_data_for_api( $data ) {
+
+		$all_sizes_data = [ ];
+
+		// Full is the only guaranteed size, so it's going to be our default
+		$size_data = wp_get_attachment_image_src( $data, 'full' );
+
+		// Something went wrong. Most likely the attachment was deleted.
+		if ( $size_data === false ) {
+			return new \stdClass;
```

Comment:
Do not just return an empty object here: if you take the pain to return something client code can still use set it up properly by filling its fields (`url`, `width` and so on) with placeholder values (e.g. empty strings or zeros).
Else just return `false` and let the client code handle the case.

Reply:
done